timer/scheduler.go

```diff
@@ -0,0 +1,63 @@
+package timer
+
+import (
+	"time"
+
+	"github.com/nivista/steady/.gen/protos/common"
+	rcron "github.com/robfig/cron/v3"
+)
+
+type cron struct {
+	sched               rcron.Schedule
+	startTime, stopTime time.Time
+	maxExecutions       int
+}
+
+var parser = rcron.NewParser(rcron.Minute | rcron.Hour | rcron.Dom | rcron.Month | rcron.Dow | rcron.SecondOptional | rcron.Descriptor)
+
+// infiniteExecutions indicates a timer should fire until cancelled.
+const infiniteExecutions = 0
+
+func (c cron) schedule(prog Progress, now time.Time) (nextFire time.Time, done bool) {
+	if c.maxExecutions != infiniteExecutions && prog.CompletedExecutions >= c.maxExecutions {
+		done = true
+		return
+	}
+
+	if prog.LastExecution.IsZero() {
+		// subtract one second so sched.Next will return startTime if it is valid
+		nextFire = c.startTime.Add(-1 * time.Second)
+	} else {
+		nextFire = prog.LastExecution
```

Comment:
I am confused - why are we assigning nextFire to the time of the last execution?

Reply:
So we can advance it to see what the next execution is. From you're comments I can see that this section needs a lot of work to be more readable (and correct). This will be addressed.